pinot-broker/src/main/java/org/apache/pinot/broker/queryquota/HelixExternalViewBasedQueryQuotaManager.java

```diff
@@ -46,21 +55,31 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import static org.apache.pinot.spi.utils.CommonConstants.Helix.DATABASE_QUERY_RATE_LIMIT;
+
 
 /**
  * This class is to support the qps quota feature.
- * It depends on the broker source change to update the dynamic rate limit,
- *  which means it only gets updated when a new table added or a broker restarted.
+ * It allows performing qps quota check at table level and database level
+ * For table level check it depends on the broker source change to update the dynamic rate limit,
+ *  which means it gets updated when a new table added or a broker restarted.
+ * For database level check it depends on the broker as well as cluster config change to update the dynamic rate limit,
+ *  which means it gets updated when the default query quota at cluster config is updated or a broker restarted.
  */
 public class HelixExternalViewBasedQueryQuotaManager implements ClusterChangeHandler, QueryQuotaManager {
```

Comment:
Do you plan to add tests? I see that there is a corresponding test class.

Reply:
Yes I will be adding test cases shortly.